src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -1070,7 +1099,11 @@ public static double GetNumericValue(Rune value)
             else
             {
                 // not an ASCII char; fall back to globalization table
+#if SYSTEM_PRIVATE_CORELIB
                 return CharUnicodeInfo.GetNumericValue(value.Value);
+#else
+                return CharUnicodeInfo.GetNumericValue(value.ToString(), 0);
```

Comment:
If this value is within the BMP, we should call the overload `CharUnicodeInfo.GetNumericValue(char)` instead. That will avoid the allocation.

Reply:
Good call, will fix.